### PR TITLE
File component: list multiple files and allow for download #1446

### PR DESCRIPTION
**Pull request by dawoodkhan82**
# Description
List each file individually and allow for download in a scroll view.

Before:
Screen Shot 2022-07-08 at 9 12 40 AM

After:
Screen Shot 2022-07-11 at 2 38 44 PM

Closes: #1446

# Checklist:

- [ ] I have performed a self-review of my own code
- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by abidlabs** [comment]
Not a lot of documentation on the issue or PR @dawoodkhan82?

**Comment by dawoodkhan82** [comment]
@abidlabs I think the multiple file component could use a legit re-design (kinda basic right now), but this pr fixes the issue where the files weren't listed nor able to download.

File: demo/zip_two_files/run.py [hunk 1]
```diff
@@ -13,7 +13,7 @@ def zip_two_files(file1, file2):
 
 demo = gr.Interface(
     zip_two_files,
-    ["file", "file"],
+    [gr.File(file_count="multiple"), gr.File(file_count="multiple")],
```
Comment (abidlabs) [inline demo/zip_two_files/run.py:16]: Why did we change both of the input components here to accept multiple files each? We should either have a single component that can accept multiple files (and change the name of this demo) or just change it back to two file components that each accept a single file

File: ui/pnpm-lock.yaml [hunk 2]
```diff
@@ -41,7 +41,7 @@ importers:
       '@tailwindcss/forms': 0.5.0_tailwindcss@3.0.23
       '@testing-library/dom': 8.11.3
       '@testing-library/svelte': 3.1.0_svelte@3.47.0
-      '@testing-library/user-event': 13.5.0_gzufz4q333be4gqfrvipwvqt6a
+      '@testing-library/user-event': 13.5.0_@testing-library+dom@8.11.3
```
Comment (abidlabs) [inline ui/pnpm-lock.yaml:44]: Why do we have all these changes in `ui/pnpm-lock.yaml`?

File: demo/zip_two_files/run.py (hunk 1 above)
Comment (dawoodkhan82) [reply demo/zip_two_files/run.py:16]: Pushed this by accident. Changed this for testing purposes

**Comment by dawoodkhan82** [comment]
> Added some thoughts above. Also, I notice that in the original component, the columns are distributed evenly:
> 
> image
> 
> In the redesigned version, the columns are all bunched towards the left:
> 
> image
> 
> Can we fix that so that the columns are distributed evenly again?
> 
> Would be great if you can give this a review as well @pngwn since I can't really comment on the frontend changes

fixed. see new screenshot

File: ui/pnpm-lock.yaml (hunk 2 above)
Comment (dawoodkhan82) [reply ui/pnpm-lock.yaml]: fixed

**Review by pngwn — APPROVED** [review]
lgtm!

**Comment by Zengyf-CVer** [comment]
@dawoodkhan82 
Can you provide the code for this example?

**Comment by dawoodkhan82** [comment]
> @dawoodkhan82 Can you provide the code for this example?

To use the multiple file input and out components you can specify `gradio.File(file_count="multiple")`

**Comment by Zengyf-CVer** [comment]
@dawoodkhan82 
Thank you very much!

**Pull request merged**